PackageReference to contentFiles pkg breaks compilation of 'net45x' tfms

Microsoft.CSharp.Core.targets is picking up a file the contentFiles section of a nuget package and causing the build to fail with this error.

> C:\Program Files\dotnet\sdk\1.0.0-preview4-004233\Microsoft.CSharp.Core.targets(71,5): error MSB3105: The item "obj\Debug\net45\NuGet\B61134ECB5E801948C09D13980B4CFE1141BB877\Microsoft.Extensions.HashCodeCombiner.Sources\1.2.0-preview1-22911" was specified more than once in the "Sources" parameter.  Duplicate items are not supported by the "Sources" parameter. [C:\dev\Universe\FileSystem\src\Microsoft.Extensions.FileSystemGlobbing\Microsoft.Extensions.FileSystemGlobbing.csproj]

**Repro:**
Use this feed to get the NuGet package in question: https://dotnet.myget.org/F/aspnetcore-ci-dev/api/v3/index.json

```xml
<Project Sdk="Microsoft.NET.Sdk" ToolsVersion="15.0">

  <PropertyGroup>
    <TargetFrameworks>net45;netstandard1.3</TargetFrameworks>
  </PropertyGroup>

  <ItemGroup>
    <Compile Include="**\*.cs" />
    <EmbeddedResource Include="**\*.resx" />
  </ItemGroup>

  <ItemGroup>
    <PackageReference Include="Microsoft.Extensions.HashCodeCombiner.Sources" Version="1.2.0-preview1-22911" PrivateAssets="All" />
    <PackageReference Include="NETStandard.Library" Version="1.6.1" />
  </ItemGroup>

  <ItemGroup Condition=" '$(TargetFramework)' == 'net45' ">
    <Reference Include="System" />
    <Reference Include="Microsoft.CSharp" />
  </ItemGroup>

</Project>
```

I'm using 1.0.0-preview4-004233.

Possibly related to https://github.com/NuGet/Home/issues/3683

--- a/src/Tasks/Microsoft.NET.Build.Tasks/ProduceContentAssets.cs
+++ b/src/Tasks/Microsoft.NET.Build.Tasks/ProduceContentAssets.cs
@@ -214,7 +214,7 @@ namespace Microsoft.NET.Build.Tasks
         }
 
         private bool IsPreprocessorFile(ITaskItem contentFile) =>
-            contentFile.GetMetadata(PPOutputPathKey) != null;
+            !string.IsNullOrEmpty(contentFile.GetMetadata(PPOutputPathKey));
 
         private void ProduceContentAsset(ITaskItem contentFile)
         {
@@ -229,7 +229,7 @@ namespace Microsoft.NET.Build.Tasks
             string ppOutputPath = contentFile.GetMetadata(PPOutputPathKey);
             string parentPackage = contentFile.GetMetadata(MetadataKeys.ParentPackage);
 
-            if (ppOutputPath != null)
+            if (!string.IsNullOrEmpty(ppOutputPath))
             {
                 if (string.IsNullOrEmpty(ContentPreprocessorOutputDirectory))
                 {
@@ -251,9 +251,10 @@ namespace Microsoft.NET.Build.Tasks
 
             if (contentFile.GetBooleanMetadata("copyToOutput") == true)
             {
-                string outputPath = contentFile.GetMetadata("outputPath") ?? ppOutputPath;
+                string outputPath = contentFile.GetMetadata("outputPath");
+                outputPath = string.IsNullOrEmpty(outputPath) ? ppOutputPath : outputPath;
 
-                if (outputPath != null)
+                if (!string.IsNullOrEmpty(outputPath))
                 {
                     var item = new TaskItem(pathToFinalAsset);
                     item.SetMetadata("TargetPath", outputPath);

--- a/src/Tasks/Microsoft.NET.Build.Tasks/build/Microsoft.PackageDependencyResolution.targets
+++ b/src/Tasks/Microsoft.NET.Build.Tasks/build/Microsoft.PackageDependencyResolution.targets
@@ -208,9 +208,9 @@ Copyright (c) .NET Foundation. All rights reserved.
 
   <Target Name="RunProduceContentAssets"
           Returns="_ContentCopyLocalItems;FileWrites"
-          DependsOnTargets="RunResolvePackageDependencies">
+          DependsOnTargets="_ComputeActiveTFMFileDependencies">
     <ItemGroup>
-      <_ContentFileDeps Include="@(FileDependencies->WithMetadataValue('FileGroup', 'ContentFile'))" />
+      <_ContentFileDeps Include="@(_ActiveTFMFileDependencies->WithMetadataValue('FileGroup', 'ContentFile'))" />
       <__ContentFileDefs Include="@(FileDefinitions)" Exclude="@(_ContentFileDeps)" />
       <_ContentFileDefs Include="@(FileDefinitions)" Exclude="@(__ContentFileDefs)" />
     </ItemGroup>

